### PR TITLE
ENH: Add unit tests for disabling top and bottom tiles

### PR DESCRIPTION
**Pull request by kenza-bouzid**

### CODE REVIEW

File: hi-ml-histopathology/src/histopathology/utils/top_bottom_tiles_utils.py [hunk 1]
```diff
@@ -364,12 +369,15 @@ def save_top_and_bottom_tiles(self, figures_dir: Path) -> None:
...
-            for slide_node in self.top_slides_heaps[class_id]:
-                case = "TN" if class_id == 0 else f"TP_{class_id}"
-                self.plot_slide_node_attention_tiles(case, figures_dir, slide_node)
-
-            for slide_node in self.bottom_slides_heaps[class_id]:
-                case = "FP" if class_id == 0 else f"FN_{class_id}"
-                self.plot_slide_node_attention_tiles(case, figures_dir, slide_node)
+        if self.num_top_slides > 0:
+            logging.info(
+                f"Plotting {self.num_top_tiles} top and bottom tiles of {self.num_top_slides} top and slides..."
```
Comment (dccastro) [inline hi-ml-histopathology/src/histopathology/utils/top_bottom_tiles_utils.py:374]: [Minor] Typo: "top and bottom slides"

File: hi-ml-histopathology/testhisto/testhisto/utils/test_top_bottom_utils.py [hunk 2]
```diff
@@ -330,3 +331,59 @@ def test_plot_top_bottom_tiles(slide_node: SlideNode, test_output_dirs: OutputFo
...
+
+    with patch.object(handler, "_update_label_slides") as mock_update_label_slides:
+        for i in range(rank * n_batches, (rank + 1) * n_batches):
+            batch_data = _batch_data(data, batch_idx=i, batch_size=batch_size)
+            batch_results = _batch_data(results, batch_idx=i, batch_size=batch_size)
+            handler.update_slides_selection(batch_data, batch_results)
+    mock_update_label_slides.assert_not_called()
+
+    for class_id in range(n_classes):
+        assert len(handler.top_slides_heaps[class_id]) == 0
```
Comment (dccastro) [inline hi-ml-histopathology/testhisto/testhisto/utils/test_top_bottom_utils.py:372]: [Minor] Possibly bottom heap as well, for good measure?

**Review by dccastro — APPROVED** [review]
Looks good, thanks for adding checks for this scenario. One minor suggestion is to log an informational message in the handler constructor when these outputs are disabled.

File: hi-ml-histopathology/src/histopathology/utils/top_bottom_tiles_utils.py [hunk 3]
```diff
@@ -133,6 +133,10 @@ def __init__(self, n_classes: int, num_top_slides: int = 10, num_top_tiles: int
         self.bottom_slides_heaps: SlideDict = {class_id: [] for class_id in range(self.n_classes)}
         self.report_cases_slide_ids = self.init_report_cases()
 
+        assert (num_top_slides >= 0 and num_top_tiles > 0) or (num_top_slides == 0), "You should use `num_top_tiles>0` "
```
[the remark this answers is not in the document]
Comment (kenza-bouzid) [reply hi-ml-histopathology/src/histopathology/utils/top_bottom_tiles_utils.py:136]: Will do!

File: hi-ml-histopathology/testhisto/testhisto/utils/test_top_bottom_utils.py (hunk 2 above)
Comment (kenza-bouzid) [reply hi-ml-histopathology/testhisto/testhisto/utils/test_top_bottom_utils.py:372]: Sure!

**Review by harshita-s — APPROVED** [review]

**Pull request merged**